tests/QE/test_xattrs.py

```diff
@@ -0,0 +1,238 @@
+import asyncio
+from pathlib import Path
+from typing import Any
+
+import pytest
+from cbltest import CBLPyTest
+from cbltest.api.cbltestclass import CBLTestClass
+from cbltest.api.error import CblSyncGatewayBadResponseError
+from cbltest.api.syncgateway import DocumentUpdateEntry, PutDatabasePayload
+
+
+@pytest.mark.sgw
+@pytest.mark.min_test_servers(0)
+@pytest.mark.min_sync_gateways(1)
+@pytest.mark.min_couchbase_servers(1)
+class TestXattrs(CBLTestClass):
+    @pytest.mark.asyncio(loop_scope="session")
+    async def test_offline_processing_of_external_updates(
+        self, cblpytest: CBLPyTest, dataset_path: Path
+    ) -> None:
+        sg = cblpytest.sync_gateways[0]
+        cbs = cblpytest.couchbase_servers[0]
+        num_docs = 1000
+        sg_db = "db"
+        bucket_name = "data-bucket"
+
+        self.mark_test_step("Create bucket and default collection")
+        cbs.drop_bucket(
+            bucket_name
+        )  # in case the bucket already exists, due to a previous failed test
+        cbs.create_bucket(bucket_name)
+
+        self.mark_test_step("Configure Sync Gateway database endpoint")
+        db_config = {
+            "bucket": bucket_name,
+            "num_index_replicas": 0,
+            "enable_shared_bucket_access": True,
+            "scopes": {"_default": {"collections": {"_default": {}}}},
```

Comment:
One thing to note here is that the default sync fn (since it isn't specified) is

```
function(doc){channel(doc.channels);}
```

but if you use a non default collection it will be

```
function (doc, oldDoc, meta) {
   channel(CollectionName);
}
```

so if you change this code to use collections it would have different behavior. I'd be inclined to specify a sync function for future use so it would be obvious if the code changed when someone went to use a non default collection.

https://docs.couchbase.com/sync-gateway/current/access-control/sync-function/sync-function.html#ex-sync-function